processing/src/main/java/io/druid/query/topn/AlphaNumericTopNMetricSpec.java

```diff
@@ -186,7 +186,8 @@ public AlphaNumericTopNMetricSpec(
       @JsonProperty("previousStop") String previousStop
   )
   {
-    super(previousStop);
+    // Set empty string as previousStop of super class otherwise we will get NPE
+    super((previousStop == null) ? "" : previousStop);
     this.previousStop = (previousStop == null) ? "" : previousStop;
```

Comment:
Actually the NPE is thrown in the comparator of AlphaNumericTopNMetricSpec. I was thinking if checking for null value there would be a better fix ? LexicographicTopNMetricSpec class (which is the super class here) does not have this problem as it checks for null values in its comparator.
The compare method of comparator in AlphaNumericTopNMetricSpec class will have a check like this -
```
      if (str1 == null) {
        return str2 == null ? 0 : -1;
      } else if (str2 == null) {
        return 1;
      } else if (str1.length() == 0) {
        return str2.length() == 0 ? 0 : -1;
      } else if (str2.length() == 0) {
        return 1;
      }
```
null check for str1 can be removed if we are sure that dimName can never be null...Any thoughts ?